src/geo/projection/globe_util.js

```diff
@@ -292,11 +292,25 @@ export function aabbForTileOnGlobe(tr: Transform, numTiles: number, tileId: Cano
     const arcB = corners[(closestArcIdx + 1) % 4];
 
     const closestArc = new Arc(arcA, arcB, arcCenter);
-    const arcBounds = [
+    let arcBounds = [
         (localExtremum(closestArc, 0) || arcA[0]),
         (localExtremum(closestArc, 1) || arcA[1]),
         (localExtremum(closestArc, 2) || arcA[2])];
 
+    // Interpolate the four corners towards their world space location in mercator projection during transition.
+    const phase = globeToMercatorTransition(tr.zoom);
+    if (phase > 0.0 && phase < 1.0) {
+        const center = [mercatorXfromLng(tr.center.lng), mercatorYfromLat(centerLat)];
```

Comment:
Some of these values are already calculated earlier, line 267:
```js
    let dx = mercatorXfromLng(tr.center.lng) - mercatorXfromLng(tileCenter.lng);
    const dy = mercatorYfromLat(centerLat) - mercatorYfromLat(tileCenterLat);
```
You could extract these as named variable to prevent redundant calls to `mercatorXfromLng` and `mercatorYfromLat`?

Also, possibly directly pass it to as two values, such as:
```js
mercatorTileCornersInCameraSpace(tileId, numTiles, tr._pixelsPerMercatorPixel, centerX, centerY);
```
So you don't need that variable anymore.